File: Assets/Scripts/Progression/SkillTreeNode.cs
using UnityEngine;
using System.Collections.Generic;

namespace ActionGameDemo.Progression
{
    [CreateAssetMenu(fileName = "SkillTreeNode", menuName = "ActionGameDemo/Skill Tree Node")]
    public class SkillTreeNodeData : ScriptableObject
    {
        [Tooltip("节点 ID")] public string nodeId = "node_0";
        [Tooltip("节点名称")] public string displayName = "New Skill";
        [Tooltip("描述")] [TextArea] public string description = "";
        [Tooltip("图标")] public Sprite icon;
        [Tooltip("所需等级")] public int requiredLevel = 1;
        [Tooltip("消耗技能点")] public int cost = 1;
        [Tooltip("前置节点 ID 列表")] public List<string> prerequisiteNodeIds = new List<string>();
        [Tooltip("效果描述")] public string effectDescription = "";

        [Header("效果")]
        public float attackBonusPercent = 0f;
        public float defenseBonusPercent = 0f;
        public float healthBonusPercent = 0f;
        public float criticalRateBonus = 0f;
    }
}

File: server/src/main/java/com/actiongame/server/domain/progression/SkillTreeNode.java
package com.actiongame.server.domain.progression;

import java.util.List;

/**
 * 技能树节点 (服务端权威)
 * 对应Unity SkillTreeNodeData, 纯数据
 */
public class SkillTreeNode {
    private final String nodeId;
    private final String displayName;
    private final String description;
    private final int requiredLevel;
    private final int cost;
    private final List<String> prerequisiteNodeIds;
    private final float attackBonusPercent;
    private final float defenseBonusPercent;
    private final float healthBonusPercent;
    private final float criticalRateBonus;

    public SkillTreeNode(String nodeId, String displayName, String description,
                         int requiredLevel, int cost, List<String> prerequisiteNodeIds,
                         float attackBonusPercent, float defenseBonusPercent,
                         float healthBonusPercent, float criticalRateBonus) {
        this.nodeId = nodeId;
        this.displayName = displayName;
        this.description = description;
        this.requiredLevel = requiredLevel;
        this.cost = cost;
        this.prerequisiteNodeIds = prerequisiteNodeIds;
        this.attackBonusPercent = attackBonusPercent;
        this.defenseBonusPercent = defenseBonusPercent;
        this.healthBonusPercent = healthBonusPercent;
        this.criticalRateBonus = criticalRateBonus;
    }

    public String getNodeId() { return nodeId; }
    public String getDisplayName() { return displayName; }
    public String getDescription() { return description; }
    public int getRequiredLevel() { return requiredLevel; }
    public int getCost() { return cost; }
    public List<String> getPrerequisiteNodeIds() { return prerequisiteNodeIds; }
    public float getAttackBonusPercent() { return attackBonusPercent; }
    public float getDefenseBonusPercent() { return defenseBonusPercent; }
    public float getHealthBonusPercent() { return healthBonusPercent; }
    public float getCriticalRateBonus() { return criticalRateBonus; }
}
